lib/std/tz.zig

```diff
@@ -0,0 +1,218 @@
+const std = @import("std.zig");
+const builtin = @import("builtin");
+
+pub const Transition = struct {
+    ts: i64,
+    timetype: *Timetype,
+};
+
+pub const Timetype = struct {
+    offset: i32,
+    flags: u8,
+    name_data: [6:0]u8,
+
+    pub fn name(self: Timetype) [:0]const u8 {
+        return std.mem.sliceTo(self.name_data[0..], 0);
+    }
+
+    pub fn isDst(self: Timetype) bool {
+        return (self.flags & 0x01) > 0;
+    }
+
+    pub fn standardTimeIndicator(self: Timetype) bool {
+        return (self.flags & 0x02) > 0;
+    }
+
+    pub fn utIndicator(self: Timetype) bool {
+        return (self.flags & 0x04) > 0;
+    }
+};
+
+pub const Leapsecond = struct {
+    occurrence: i48,
+    correction: i16,
+};
+
+pub const Tz = struct {
+    allocator: std.mem.Allocator,
+    transitions: []const Transition,
+    timetypes: []const Timetype,
+    leapseconds: []const Leapsecond,
+    footer: []const u8,
+
+    pub fn parse(allocator: std.mem.Allocator, data: []const u8) !Tz {
+        const header_size = 4 + 1 + 15 + 6 * 4;
+        if (data.len < header_size) return error.BadSize;
+
+        const magic_l = data[0..4];
+        const version_l = data[4];
+        if (!std.mem.eql(u8, magic_l, "TZif")) return error.BadHeader;
+        if (version_l != '2' and version_l != '3') return error.BadVersion;
+
+        // Parse the legacy header and skip the entire thing
+        const isutcnt_l = std.mem.readIntBig(u32, data[20..24]);
+        const isstdcnt_l = std.mem.readIntBig(u32, data[24..28]);
+        const leapcnt_l = std.mem.readIntBig(u32, data[28..32]);
+        const timecnt_l = std.mem.readIntBig(u32, data[32..36]);
+        const typecnt_l = std.mem.readIntBig(u32, data[36..40]);
+        const charcnt_l = std.mem.readIntBig(u32, data[40..44]);
+        const data_block_size_legacy = timecnt_l * 5 + typecnt_l * 6 + charcnt_l + leapcnt_l * 8 + isstdcnt_l + isutcnt_l;
+        if (data.len < header_size + data_block_size_legacy) return error.BadSize;
+
+        const data2 = data[header_size + data_block_size_legacy ..];
+        if (data2.len < header_size) return error.BadSize;
+
+        const magic = data2[0..4];
+        const version = data2[4];
+        if (!std.mem.eql(u8, magic, "TZif")) return error.BadHeader;
+        if (version != '2' and version != '3') return error.BadVersion;
+
```

Comment:
Should verify that it is followed by 15 nulls

Reply:
RFC8536 doesn't actually say that they should be null, just that they are reserved.

It's even implied that you shouldn't care about them - the spec depends on that legacy parsers that doesn't understand the "version" field not bailing when it sees a '2' or '3' in that byte, otherwise the "fat" version of the files wouldn't work.